tests/models/test_popular.py

```diff
@@ -212,3 +219,146 @@ def test_i2i(
     def test_second_fit_refits_model(self, dataset: Dataset) -> None:
         model = PopularModel()
         assert_second_fit_refits_model(model, dataset)
+
+
+class TestPopularModelConfiguration:
+    @pytest.mark.parametrize("begin_from", (None, datetime(2021, 11, 23), "2021-11-23T10:20:30.400+02:30"))
+    @pytest.mark.parametrize(
+        "period",
+        (
+            None,
+            timedelta(days=7),
+            {
+                "days": 7,
+                "seconds": 123,
+                "microseconds": 12345,
+                "milliseconds": 32,
+                "minutes": 2,
+                "weeks": 7,
+            },
+        ),
+    )
+    def test_from_config(
+        self, period: tp.Optional[tp.Union[timedelta, dict]], begin_from: tp.Optional[tp.Union[datetime, str]]
+    ) -> None:
+        config = {
+            "popularity": "n_interactions",
+            "period": period,
+            "begin_from": begin_from,
+            "add_cold": True,
+            "inverse": True,
+            "verbose": 0,
+        }
+        if period is not None and begin_from is not None:
+            with pytest.raises(ValueError):
```

Comment:
this shouldn't be here, it's a class to test configs, not parameter validation

Reply:
I think in this case it's better not to use 2 parameterize blocks, and put everything into single one:  
you have 3 options in each so 9 combinations, but actually you run only 5.

Reply:
Fixed tests:
- ValueError tests case was removed;
- expected values were moved to parametrize.